btcec/signature.go

```diff
@@ -307,10 +340,10 @@ func recoverKeyFromSignature(curve *KoblitzCurve, sig *Signature, msg []byte,
 	// We calculate the two terms sR and eG separately multiplied by the
 	// inverse of r (from the signature). We then add them to calculate
 	// Q = r^-1(sR-eG)
-	invr := new(big.Int).ModInverse(sig.R, curve.Params().N)
+	invr := new(big.Int).ModInverse(sigR, curve.Params().N)
 
 	// first term.
-	invrS := new(big.Int).Mul(invr, sig.S)
+	invrS := new(big.Int).Mul(invr, sigS)
```

Comment:
Same thing here.  This math should be done in the field where it's faster and convert the results to `big.Int`.